bindings/experimental/distrdf/python/DistRDF/Backends/Base.py

```diff
@@ -157,18 +156,25 @@ def mapper(current_range):
             # environment
             initialization()
 
-            # Build rdf
-            start = int(current_range.start)
-            end = int(current_range.end)
-
             if treename is not None:
+                # Build TEntryList for this range:
+                elists = ROOT.TEntryList()
+
                 # Build TChain of files for this range:
                 chain = ROOT.TChain(treename)
-                for f in current_range.filelist:
-                    chain.Add(str(f))
+                for start, end, filename in zip(current_range.localstarts, current_range.localends,
+                                                current_range.filelist):
+                    elist = ROOT.TEntryList("e","e", treename, filename)
+                    for entry in range(start, end):
+                        elist.Enter(entry)
+                    elists.Add(elist)
+                    chain.Add(filename)
 
                 # We assume 'end' is exclusive
-                chain.SetCacheEntryRange(start, end)
+                chain.SetCacheEntryRange(current_range.globalstart, current_range.globalend)
```

Comment:
This is global for this range, right? (for the mini chain)

Reply:
Yes exactly! Thus the need for keeping `globalstart` and `globalend` in the range object

Reply:
I'm thinking then that `globalstart` can be misleading. Should it be `chain_start`? Or something better...

Reply:
`current_range.chainfirstentry` and `current_range.chainlastentry` ?

Reply:
Never mind, like I said below we will eventually not need them anyway.